docs/design/proposals/VRG-Reconciliation.md

```diff
@@ -0,0 +1,150 @@
+# VRG reconciler workflow
+
+This design document covers the VRG CRD and its reconciliation workflow.
+
+**NOTE**: This is added to aid understanding and ease development of the
+various parts in parallel (where possible) and would possibly be merged once
+it reaches initial completion.
+
+## Terminology
+
+- VRG: VolumeReplicationGroup
+- VR: VolumeReplication
+- PVC: PersistentVolumeClaim
+- PV: PersistentVolume
+
+## VRG CRD
+
+- Spec:
+    - PVCSelector
+        - Label selector to identify all the PVCs that are in this group that
+        needs to be replicated to the peer cluster.
+    - VolumeReplicationClass
+        - ReplicationClass of all volumes in this replication group; this
+        value is propagated to children VolumeReplication CRs
+    - ReplicationState
+        - Desired state of all volumes [`primary` or `secondary`] in this
+        replication group; this value is propagated to children
+        VolumeReplication CRs
+    - S3Endpoint
+        - S3 Endpoint to replicate PV metadata
+    - S3SecretName
+        - S3 Endpoint to replicate PV metadata
+- Status:
+    - **TBD**
+
+### VRG Example
+
+```
+---
+apiVersion: ramendr.openshift.io/v1alpha1
+kind: VolumeReplicationGroup
+metadata:
+  name: vrg-sample
+spec:
+  pvcSelector:
+    matchLabels:
+      appname: myapp
+  volumeReplicationClass: "vr-class-sample"
+  replicationState: "Primary"
+  s3Endpoint: http://objectstore.svc:9000
+  s3SecretName: vrg-objectstore-secret
+```
+
+## Reconciling VRG as Primary
+
+1. Select all PVCs that match `PVCSelector` for processing
+1. For each PVC
+    1. If `phase` is not [bound](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#phase),
+    continue but requeue
+    1. Check `vr-protected` finalizer, if found,
+        1. Check and remove (stale) `vr-protecting` finalizer
+        1. Check and update VR as `primary` and continue
+    1. Add `vr-protecting` finalizer to PVC, to order VR deletion before PVC is deleted
```

Comment:
Similar to "Check `vr-protected` finalizer", should this also be phrased "Check `vr-protecting` finalizer" and do some action if not found?